src/expressions/operators/compares/Compare.ts

```diff
@@ -75,12 +157,41 @@ class Compare extends Expression {
 					},
 					default: () => {
 						if (this._compare === 'nodeCompare') {
-							return nodeCompare(
-								this._operator,
-								executionParameters.domFacade,
-								firstSequence,
-								secondSequence
-							);
+							return firstSequence.switchCases({
```

Comment:
Oh wow, this is getting deep. Can you try to split up this function in some logical parts to make it more readable?

Reply:
Extracted these lines to a new function to reduce the depth of this bumpy hill up and down